rectools/models/nn/transformer_base.py

```diff
@@ -451,95 +462,257 @@ def _xavier_normal_init(self) -> None:
-        self.fit_trainer: Trainer
+        self.get_val_mask_func = get_val_mask_func
+
+        self._init_torch_model()
+        self._init_data_preparator()
+
         if trainer is None:
-            self._trainer = Trainer(
-                max_epochs=epochs,
-                min_epochs=epochs,
-                deterministic=deterministic,
-                enable_progress_bar=verbose > 0,
-                enable_model_summary=verbose > 0,
-                logger=verbose > 0,
-            )
+            self._init_trainer()
         else:
             self._trainer = trainer
+
+        self.lightning_model: SessionEncoderLightningModuleBase
         self.data_preparator: SessionEncoderDataPreparatorBase
-        self.u2i_dist = Distance.DOT
-        self.i2i_dist = Distance.COSINE
-        self.lr = lr
-        self.loss = loss
-        self.gbce_t = gbce_t
+        self.fit_trainer: Trainer
 
-    def _fit(
-        self,
-        dataset: Dataset,
-    ) -> None:
-        self.data_preparator.process_dataset_train(dataset)
-        train_dataloader = self.data_preparator.get_dataloader_train()
-        val_dataloader = self.data_preparator.get_dataloader_val()
+    def _check_devices(self, recommend_devices: tp.Union[int, tp.List[int]]) -> None:
+        if isinstance(recommend_devices, int) and recommend_devices != 1:
+            raise ValueError("Only single device is supported for inference")
+        if isinstance(recommend_devices, list) and len(recommend_devices) > 1:
+            raise ValueError("Only single device is supported for inference")
 
-        torch_model = deepcopy(self._torch_model)
-        torch_model.construct_item_net(self.data_preparator.train_dataset)
+    def _init_data_preparator(self) -> None:
+        raise NotImplementedError()
+
+    def _init_trainer(self) -> None:
+        self._trainer = Trainer(
+            max_epochs=self.epochs,
+            min_epochs=self.epochs,
+            deterministic=self.deterministic,
+            enable_progress_bar=self.verbose > 0,
+            enable_model_summary=self.verbose > 0,
+            logger=self.verbose > 0,
+            enable_checkpointing=False,
+            devices=1,
+        )
 
+    def _init_torch_model(self) -> None:
+        self._torch_model = TransformerBasedSessionEncoder(
+            n_blocks=self.n_blocks,
+            n_factors=self.n_factors,
+            n_heads=self.n_heads,
+            session_max_len=self.session_max_len,
+            dropout_rate=self.dropout_rate,
+            use_pos_emb=self.use_pos_emb,
+            use_causal_attn=self.use_causal_attn,
+            use_key_padding_mask=self.use_key_padding_mask,
+            transformer_layers_type=self.transformer_layers_type,
+            item_net_block_types=self.item_net_block_types,
+            pos_encoding_type=self.pos_encoding_type,
+        )
+
+    def _init_lightning_model(self, torch_model: TransformerBasedSessionEncoder, n_item_extra_tokens: int) -> None:
```

Comment:
`n_item_extra_tokens` isn't used to initialization `lightning_model` as it is passed `data_preparator`

Reply:
fixed